ui/src/app/pages/profile/data-access-panel.spec.tsx

```diff
@@ -0,0 +1,56 @@
+import {mount} from 'enzyme';
+import * as React from 'react';
+import * as TestHelpers from 'testing/react-test-helpers';
+import {DataAccessPanel} from 'app/pages/profile/data-access-panel';
+import {CheckCircle} from 'app/components/icons'
+
+
+describe('Data Access Panel', () => {
+  const component = (props = {}) => {
+    return mount(<DataAccessPanel {...props}/>);
+  };
+
+  it('Should show controlled tier message while the user institution does not have an agreement', async() => {
+    const wrapper = component({hasInstitutionalAgreement: false});
+    expect(TestHelpers.findNodesContainingText(wrapper, "your institution will need to sign").length).toBe(1)
+  });
+
+  it('Should show revoked status for controlled tier when the user has been revoked', async() => {
+    const wrapper = component({hasInstitutionalAgreement: true, userRevoked: true});
+    expect(TestHelpers.findNodesByExactText(wrapper, "Access to controlled tier data is revoked.").length).toBe(1)
+  });
+
+  it('Should show "get started" status for controlled tier when the user has not completed training', async() => {
+    const wrapper = component({hasInstitutionalAgreement: true, tiers: []});
+    expect(TestHelpers.findNodesByExactText(wrapper, "Get Started").length).toBe(1)
+  });
+
+  it('Should show success status for controlled tier when the user has completed training', async() => {
+    const wrapper = component({hasInstitutionalAgreement: true, tiers: ['controlled']});
+    expect(wrapper.find(CheckCircle).length).toBe(1);
+  });
+
+  it('Should show success status when the user is in the registered tier', async() => {
+    const wrapper = component({hasInstitutionalAgreement: true, tiers: ['registered']});
+    expect(wrapper.find(CheckCircle).length).toBe(1);
+  });
+
+  it('Should show success status when the user is in the registered tier and controlled tier', async() => {
+    const wrapper = component({hasInstitutionalAgreement: true, tiers: ['registered', 'controlled']});
+    expect(wrapper.find(CheckCircle).length).toBe(2);
+  });
+
+  it('Should show success status when the user is in the registered tier and "get stated" for the controlled tier', async() => {
+    const wrapper = component({hasInstitutionalAgreement: true, tiers: ['registered']});
+    expect(wrapper.find(CheckCircle).length).toBe(1);
+    expect(TestHelpers.findNodesByExactText(wrapper, "Get Started").length).toBe(1)
+  });
+
+  it('Should not show success status when the user is in not the registered tier and "get stated" for the controlled tier', async() => {
+    const wrapper = component({hasInstitutionalAgreement: true, tiers: []});
+    expect(wrapper.find(CheckCircle).length).toBe(0);
+    expect(TestHelpers.findNodesContainingText(wrapper, 'Please complete data access').length).toBe(1);
+    expect(TestHelpers.findNodesByExactText(wrapper, "Get Started").length).toBe(1);
+  });
+
+});
```

Comment:
I think these eight tests can be reduced to just one that verifies that `Utils.cond` works as expected. My judgement is that each of these tests is redundantly asserting that claim.

Reply:
I disagree.  These high level tests are useful to show the high level behavior is working as intended, and also shows us these intentions.

Reply:
I would like to see a separate spec for cond and maybe actually

Reply:
They do seem a bit redundant. The intention is to test that the component renders correctly depending on the props passed in. I know it is being controlled by the cond, but this will also test that the correct information is rendered based on the state of the component. I tried doing this with setProps in enzyme to make it a bit more concise, but it seems like enzyme does not like to actually set the props and update the component

Reply:
I added a test for cond

Reply:
The logic here is too close to the logic in the code for my comfort. Perhaps just an example of a bug these tests might catch? @jmthibault79 to your point, I judge that if I can trust `Util.cond`, I can verify the high-level behavior by inspection of the code below. I do appreciate the descriptive nature of the test code to serve this purpose, but not enough to justify the additional code.

I've said my piece. Whatever you choose to do is the right choice.